app/allocations/middleware.js

```diff
@@ -1,44 +1,149 @@
 const { format } = require('date-fns')
+const { cloneDeep, isEqual, mapValues } = require('lodash')
+const queryString = require('query-string')
 
 const {
   dateFormat,
   getDateFromParams,
   getPeriod,
 } = require('../../common/helpers/date-utils')
+const presenters = require('../../common/presenters')
 const allocationService = require('../../common/services/allocation')
 
-const allocationsMiddleware = {
-  // TODO: refactor this to be common with the /moves pagination.
-  //  It differs only in the output of the url
-  setPagination: (req, res, next) => {
-    const { period } = req.params
-    const today = format(new Date(), dateFormat)
-    const baseDate = getDateFromParams(req)
-    const interval = period === 'week' ? 7 : 1
+const allocationTypeNavigationConfig = [
+  {
+    label: 'allocations::complete',
+    filter: { complete_in_full: true },
+  },
+  {
+    label: 'allocations::incomplete',
+    filter: { complete_in_full: false },
+  },
+]
 
-    const previousPeriod = getPeriod(baseDate, -interval)
-    const nextPeriod = getPeriod(baseDate, interval)
+// TODO: refactor this to be common with the /moves pagination.
```

Comment:
Is there any reason not to hoist `setPagination` out of /moves and share it?

Reply:
They differs in the way they build the urls for prev/next/today, because the urls are different. I toyed with a sort of factory pattern that passed to the factory the schema of the url, but in fact it didn't save any code, and it made it more complicated. 
It remains certainly a todo, but I'd rather not tackle it in this PR.